### ISSUE TITLE
radio as boolean (0, 1) and parseNumbers

### ISSUE THREAD

**Issue by oleg-andreyev** [react-querybuilder/react-querybuilder#622]
Right now experiencing issue with radio element, which holds boolean (0, 1).

![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/b59b9b0f-951d-4798-9a98-cdfa332d11ff)
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/c544d7f9-310b-4e3e-8a25-80ef8322f89a)
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/75c9bf62-fb79-46d9-9fa0-367e17f0ab6a)
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/4b88b177-a3d1-4dcf-9d95-2150f3523243)

As soon I save and render components once again I see
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/5a08d35b-5106-4db1-9404-ff01519b62aa)
and non-values are selected
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/48a85e31-117f-44d2-9c29-f2c59331724a)

----
**UPD**

Noticed that values `name` is string, that's why it cannot match.

![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/a60a5f98-1e23-472d-be49-d90916216d6f)
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/291552c1-ea0a-4503-b07c-dd6052d77e97)

but as soon as change type to int, types are swapped in `value`
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/5b444353-c32c-4bd8-a16d-2159f3550736)

----

initial load
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/f78af77c-57eb-4887-9002-7ad69dac54b1)
as soon I click on radio
![image](https://github.com/react-querybuilder/react-querybuilder/assets/1244112/a0f3af52-1686-4d6f-8d70-b0d26605fd62)

**Comment by oleg-andreyev** [comment]
atm applied following workaround:

```js
    if (props.type === 'radio') {
        return (
            <ValueEditor
                {...props}
                handleOnChange={(v) => {
                    let valuesType = typeof props.values[0].name;
                    let valueType = typeof v;

                    if (valueType !== valuesType && valuesType === 'number') {
                        props.handleOnChange(parseInt(v))
                    }
                }}
            />
        );
    }
```

**Comment by jakeboone02** [comment]
Thanks for the report. I think your issue might be more with the behavior of native HTML `<input>` elements, which always report their `value` as a `string`, than RQB. What would you propose as a workaround/fix?

**Comment by oleg-andreyev** [comment]
@jakeboone02 I've added a code-snippet here https://github.com/react-querybuilder/react-querybuilder/issues/622#issuecomment-1872126396, I see that we'll see to compare values and value, if values has Int, then value should be casted to Int, and vice-versa.

**Comment by jakeboone02** [comment]
Yeah, I saw that but didn't consider it might be the actual solution. Want to submit a PR with that change while I think it over ([code is here](https://github.com/react-querybuilder/react-querybuilder/blob/196ed93117a471cafa4079a7d7cc025e5bfca361/packages/react-querybuilder/src/components/ValueEditor.tsx#L136-L153))? An additional test ([radio tests here](https://github.com/react-querybuilder/react-querybuilder/blob/196ed93117a471cafa4079a7d7cc025e5bfca361/packages/react-querybuilder/genericTests/testValueEditor.tsx#L223C23-L271)) will probably be necessary to maintain 100% coverage.
